middleman_protocol/middleman_protocol/message.py

```diff
@@ -0,0 +1,302 @@
+    """ Subclass of AbstractMiddlemanMessage for handling Golem Message payload type. """
+
+    payload_type = PayloadType.GOLEM_MESSAGE
+
+    @classmethod
+    def _deserialize_payload(cls, payload: bytes) -> BaseGolemMessage:
+        return BaseGolemMessage.deserialize(payload, None, check_time=False)
+
+    def _serialize_payload(self, payload: BaseGolemMessage) -> bytes:
+        return payload.serialize()
+
+    def _validate_payload(self, payload: BaseGolemMessage) -> None:
+        if not isinstance(payload, BaseGolemMessage):
+            raise MiddlemanProtocolError(
+                f'Trying to create GolemMessageMiddlemanMessage but passed payload type is {type(payload)} '
+                f'instead of Golem Message.'
+            )
+
+
+@register
+class ErrorMiddlemanMessage(AbstractMiddlemanMessage):
+    """ Subclass of AbstractMiddlemanMessage for handling error code with error message payload type. """
+
+    payload_type = PayloadType.ERROR
+
+    @classmethod
+    def _deserialize_payload(cls, payload: bytes) -> tuple:
+        error_payload_format = cls.get_error_payload_format()
+        error_payload = error_payload_format.parse(payload)
+        return (error_payload.error_code, error_payload.error_message)
+
+    def _serialize_payload(self, payload: tuple) -> bytes:
+        error_payload_format = self.get_error_payload_format()
+        error_payload = Container(
+            error_code=payload[0],
+            error_message=payload[1],
+        )
+        raw_error_payload = error_payload_format.build(error_payload)
+        return raw_error_payload
+
+    def _validate_payload(self, payload: tuple) -> None:
+        if not isinstance(payload, tuple):
+            raise MiddlemanProtocolError(
+                f'Trying to create ErrorMiddlemanMessage but passed type payload is '
+                f'{type(payload)} instead of tuple. It must be pair of error code and error message.'
+            )
+        if len(payload) != 2:
+            raise MiddlemanProtocolError(
+                f'Trying to create ErrorMiddlemanMessage but passed payload tuple has length '
+                f'{len(payload)} instead of 2. It must be pair of error code and error message.'
+            )
+        if not isinstance(payload[0], int):
+            raise MiddlemanProtocolError(
+                f'First element of payload tuple passed to ErrorMiddlemanMessage must be error code integer '
+                f'instead of {type(payload[0])}.'
+            )
+        if not isinstance(payload[1], str):
+            raise MiddlemanProtocolError(
+                f'Second element of payload tuple passed to ErrorMiddlemanMessage must be error message string '
+                f'instead of {type(payload[1])}.'
+            )
+
+    @classmethod
+    def get_error_payload_format(cls) -> Struct:
+        return Struct(
+            error_code=Int8ub,
+            error_message=PascalString(VarInt, 'utf8'),
```

Comment:
It will simply ignore them.

```
In [1]: from construct import *
In [2]: frame = Struct(error_message=PascalString(VarInt, 'utf8'))
In [4]: frame_build = frame.build(Container(error_message=b'1'))
In [5]: frame_build = frame.build(Container(error_message='1'))
In [6]: len(frame_build)
Out[6]: 2
In [7]: frame_build
Out[7]: b'\x011'
In [8]: frame_build += b'\x001'
In [9]: len(frame_build)
Out[9]: 4
In [10]: frame.parse(frame_build)
Out[10]: Container(error_message=u'1')
```

Reply:
Is see that it raises `StreamError` when there's less than stated number of characters:

``` python
frame.parse(b'\x05ab')
```
```
construct.core.StreamError: stream read less then specified amount, expected 5, found 2
```

This exception should not crash the application. Please make a test for that case.

Also, I still think that the case with more characters than stated should have a test too.